modules/tests/shared/src/test/scala/coursier/test/VersionTests.scala

```diff
@@ -214,43 +187,57 @@ object VersionTests extends TestSuite {
 
 
     'wellKnownQualifierVersusUnknownQualifierOrdering - {
-      assert(compare("1-abc", "1-alpha" ) > 0)
-      assert(compare("1-abc", "1-beta" ) > 0)
-      assert(compare("1-abc", "1-milestone" ) > 0)
-      assert(compare("1-abc", "1-rc" ) > 0)
-      assert(compare("1-abc", "1-snapshot" ) > 0)
-      assert(compare("1-abc", "1" ) > 0)
-      assert(compare("1-abc", "1-sp" ) > 0)
+      assert(compare("1-milestone", "1-rc" ) < 0)
+      assert(compare("1-milestone", "1-beta" ) < 0)
+      assert(compare("1-M1", "1-rc1" ) < 0)
+
+      assert(compare("1-abc", "1-alpha" ) < 0)
+      assert(compare("1-abc", "1-beta" ) < 0)
+      assert(compare("1-abc", "1-milestone" ) < 0)
+      assert(compare("1-abc", "1-rc" ) < 0)
+      assert(compare("1-abc", "1-snapshot" ) < 0)
+      assert(compare("1-abc", "1" ) < 0)
+      assert(compare("1-abc", "1-sp" ) < 0)
+
+      assert(compare("1.0m", "1.0" ) < 0)
+      assert(compare("1.0-m", "1.0" ) < 0)
+      assert(compare("1.0.m", "1.0" ) < 0)
+
+      assert(compare("1.0m1", "1.0" ) < 0)
+      assert(compare("1.0-m1", "1.0" ) < 0)
+      assert(compare("1.0.m1", "1.0" ) < 0)
+      assert(compare("1.0m.1", "1.0" ) < 0)
+      assert(compare("1.0m-1", "1.0" ) < 0)
+
+      assert(compare("1.0.1-MF", "1.0.0" ) > 0)
+      assert(compare("1.0.1-MF", "1.0.1" ) < 0)
+      assert(compare("1.0.1-MF", "1.0.2" ) < 0)
+      assert(compare("1.0.1-X20", "1.0.0" ) > 0)
+      assert(compare("1.0.1-X20", "1.0.1" ) < 0)
+      assert(compare("1.0.1-X20", "1.0.2" ) < 0)
+      assert(compare("1.0.1-SNAP12", "1.0.0" ) > 0)
+      assert(compare("1.0.1-SNAP12", "1.0.1" ) < 0)
```

Comment:
> I'd edge towards educating people about version schemes… `SNAP12` looks definitely like an abuse of notation to me… Same for `-X10`. 

I think treating unknown qualifiers as pre-release is more sane way of handling the versions. See for example https://semver.org/spec/v2.0.0.html#spec-item-9:

> A pre-release version MAY be denoted by appending a hyphen and a series of dot separated identifiers immediately following the patch version. Identifiers MUST comprise only ASCII alphanumerics and hyphen [0-9A-Za-z-].